jwt/algorithms.py

```diff
@@ -44,22 +73,21 @@ def get_default_algorithms():
     }
 
     if has_crypto:
-        default_algorithms.update({
-            'RS256': RSAAlgorithm(RSAAlgorithm.SHA256),
-            'RS384': RSAAlgorithm(RSAAlgorithm.SHA384),
-            'RS512': RSAAlgorithm(RSAAlgorithm.SHA512),
-            'ES256': ECAlgorithm(ECAlgorithm.SHA256),
-            'ES384': ECAlgorithm(ECAlgorithm.SHA384),
-            'ES521': ECAlgorithm(ECAlgorithm.SHA512),
-            'ES512': ECAlgorithm(ECAlgorithm.SHA512),  # Backward compat for #219 fix
-            'PS256': RSAPSSAlgorithm(RSAPSSAlgorithm.SHA256),
-            'PS384': RSAPSSAlgorithm(RSAPSSAlgorithm.SHA384),
-            'PS512': RSAPSSAlgorithm(RSAPSSAlgorithm.SHA512)
-        })
+        crypto_algorithms = _get_crypto_algorithms()
+        default_algorithms.update(crypto_algorithms)
 
     return default_algorithms
 
 
+def get_crypto_algorithms():
```

Comment:
Instead of the `get_crypto_algorithms()` approach, I think it might be simpler to just use a set to define the cryptography-supplied algorithms.

Something like:

```
requires_cryptography = set([
    'RS256', 'RS384', 'RS512', 'ES256', 'ES384', 'ES521',
    'PS256', 'PS384', 'PS512'
])
```

Then instead of `if not has_crypto and algorithm in get_crypto_algorithms()`, you could simply check `if algorithm in requires_cryptography`.